smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/RuntimeClientPlugin.java

```diff
@@ -389,6 +428,39 @@ public Builder resolveFunction(Symbol resolveFunction) {
             return resolveFunction(SymbolReference.builder().symbol(resolveFunction).build());
         }
 
+        /**
+         * Sets the symbol that is invoked in order to convert the
+         * input symbol type to a resolved symbol type.
+         *
+         * <p>If this is set, then both {@link #resolvedConfig} and
+         * {@link #inputConfig} must also be set.
+         *
+         * @param resolveFunction Function used to convert input to resolved.
+         * @param additionalParameters Additional parameters to be generated as resolve function input.
+         * @return Returns the builder.
+         * @see #getResolveFunction()
+         */
+        public Builder resolveFunction(Symbol resolveFunction, String... additionalParameters) {
+            return resolveFunction(SymbolReference.builder().symbol(resolveFunction).build(), additionalParameters);
+        }
+
+        /**
+         * Add additional positional input parameters to resolve function
+         * <p>this can only be set if {@link #resolveFunction} is set.
+         *
+         * @param additionalParameters Additional parameters to be generated as resolve function input.
+         * @return Returns the builder.
+         * @see #getResolveFunction()
+         */
+        public Builder addAdditionalResolveFunctionParameters(String... additionalParameters) {
+            if (this.resolveFunction == null) {
```

Comment:
It's a little odd to restrict the order of functions called on a builder. This would be better as an exception thrown in the constructor if there are `additionalParameters` and the function hasn't been set.